ene_kafka/src/dispatchers/mod.rs

```diff
@@ -32,26 +32,25 @@ macro_rules! generate_handler_types_enum {
 /// The macro expects a list of handlers that will be used to dispatch the events.
 #[macro_export]
 macro_rules! generate_event_dispatcher {
-    ($($handler: ident $(< $( $generic_identifier:tt $( : $identifier_constraint:tt $(+ $identifier_additions:tt )* )? ),+ >)?),*) => {
-        ene_kafka::generate_handler_types_enum!($($handler $(< $( $generic_identifier $( : $identifier_constraint $(+ $identifier_additions )* )? ),+ >)?),*);
+    ($($handler_name: ident: $handler_type: ident $(< $( $generic_identifier:tt $( : $identifier_constraint:tt $(+ $identifier_additions:tt )* )? ),+ >)?),*) => {
+       // ene_kafka::generate_handler_types_enum!($($handler $(< $( $generic_identifier $( : $identifier_constraint $(+ $identifier_additions )* )? ),+ >)?),*);
         struct CloudEventDispatcher {
-            handlers: Vec<HandlerTypes>,
+           $(
+               $handler_name: $handler_type $(< $( $generic_identifier $( : $identifier_constraint $(+ $identifier_additions )* )? ),+ >)?,
+           )*
         }
 
 
     #[async_trait::async_trait]
     impl ene_kafka::dispatchers::EventDispatcher for CloudEventDispatcher {
 
         async fn dispatch_event<Event: ene_kafka::messages::cloud_events::cloud_event::CloudEvent<String, String>>(&self, event: &Event) -> anyhow::Result<()> {
-            for handler in self.handlers.iter() {
-                match handler {
-                    $(HandlerTypes::$handler(handler) => {
-                        if ene_kafka::handlers::EventHandler::can_handle(handler, event)? {
-                            ene_kafka::handlers::EventHandler::deserialize_and_handle(handler, event).await?;
-                        }
-                    }),*
+            use ene_kafka::handlers::EventHandler;
+            $(
+                if self.$handler_name.can_handle(event)? {
```

Comment:
This will change the semantics of how handlers no work. This will make it possible for the same event to pass through multiple handlers. I think personally that it's a nice to have, but it can also introduce subtle bugs

Reply:
Update: I changed my mind. it would be tricky to know what failed.